marketplace/src/benchmarking.rs

```diff
@@ -17,236 +17,166 @@
+	// Create default marketplace.
 	assert_ok!(Marketplace::<T>::create_marketplace(
-		get_origin::<T>("ALICE").into(),
+		alice_origin.into(),
 		MarketplaceType::Public,
-		0,
-		bounded_vec![50],
-		bounded_vec![],
-		bounded_vec![],
-		bounded_vec![],
-	));
-	let public_id = Marketplace::<T>::marketplace_id_generator();
-
-	// Create Private Marketplace for Alice
-	assert_ok!(Marketplace::<T>::create_marketplace(
-		get_origin::<T>("ALICE").into(),
-		MarketplaceType::Private,
-		0,
-		bounded_vec![51],
-		bounded_vec![],
-		bounded_vec![],
-		bounded_vec![],
+		Some(MarketplaceFee::Percentage(PERCENT_50)),
+		Some(MarketplaceFee::Percentage(PERCENT_50)),
+		Some(marketplace_offchain_data),
 	));
-	let private_id = Marketplace::<T>::marketplace_id_generator();
-
-	(public_id, private_id, nft_id)
-}
-
-pub fn get_account<T: Config>(name: &'static str) -> T::AccountId {
-	let account: T::AccountId = benchmark_account(name, 0, 0);
-	account
-}
-
-pub fn get_origin<T: Config>(name: &'static str) -> RawOrigin<T::AccountId> {
-	RawOrigin::Signed(get_account::<T>(name))
 }
 
 benchmarks! {
-	list_nft {
-		let (mkp_id, _, nft_id) = prepare_benchmarks::<T>();
-
-		let alice: T::AccountId = get_account::<T>("ALICE");
-		let price: BalanceOf<T> = 100u32.into();
-
-	}: _(RawOrigin::Signed(alice.clone()), nft_id, price, Some(mkp_id))
-	verify {
-		assert_eq!(T::NFTExt::owner(nft_id), Some(alice));
-		assert_eq!(NFTsForSale::<T>::contains_key(nft_id), true);
-	}
-
-	unlist_nft {
-		let (mkp_id, _, nft_id) = prepare_benchmarks::<T>();
-
-		let alice = get_origin::<T>("ALICE");
-		let price: BalanceOf<T> = 100u32.into();
-		drop(Marketplace::<T>::list_nft(alice.clone().into(), nft_id, price, Some(mkp_id)));
-
-	}: _(alice.clone(), nft_id)
-	verify {
-		assert_eq!(NFTsForSale::<T>::contains_key(nft_id), false);
-	}
-
-	buy_nft {
-		let (mkp_id, _, nft_id) = prepare_benchmarks::<T>();
-
-		let bob: T::AccountId = get_account::<T>("BOB");
-		let price: BalanceOf<T> = 0u32.into();
-
-		drop(Marketplace::<T>::list_nft(get_origin::<T>("ALICE").into(), nft_id, price, Some(mkp_id)));
-	}: _(RawOrigin::Signed(bob.clone().into()), nft_id)
-	verify {
-		assert_eq!(T::NFTExt::owner(nft_id), Some(bob));
-		assert_eq!(NFTsForSale::<T>::contains_key(nft_id), false);
-	}
-
 	create_marketplace {
 		prepare_benchmarks::<T>();
-
 		let alice: T::AccountId = get_account::<T>("ALICE");
-		let mkp_id = Marketplace::<T>::marketplace_id_generator() + 1;
-	}: _(RawOrigin::Signed(alice.clone().into()), MarketplaceType::Public, 0, bounded_vec![20, 30, 40], bounded_vec![], bounded_vec![], bounded_vec![])
+		let alice_origin = origin::<T>("ALICE");
+		let marketplace_offchain_data =
+			BoundedVec::try_from(vec![1; T::OffchainDataLimit::get() as usize])
+				.expect("It will never happen.");
+	}: _(alice_origin, MarketplaceType::Public, Some(MarketplaceFee::Percentage(PERCENT_50)), Some(MarketplaceFee::Percentage(PERCENT_50)), Some(marketplace_offchain_data))
```

Comment:
Instead of passing `alice_origin` you could pass `origin::<T>("ALICE"):` less code :)